crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -567,3 +710,27 @@ fn create_if_subgraph(ctx: &mut SsaContext, prev_block: BlockId) -> (BlockId, Bl
 
     (new_entry, new_exit)
 }
+
+#[derive(Debug)]
+struct Segment {
+    a: (usize, NodeId),
+    b: (usize, NodeId),
+}
+
+impl Segment {
+    pub fn new(i: (usize, &NodeId), j: (usize, &NodeId)) -> Segment {
```

Comment:
Instead of i and j, perhaps first and second or just a and b to match the field names.

Reply:
Done, with this name: left/right_node